src/sinks/datadog/metrics/encoder.rs

```diff
@@ -394,6 +440,22 @@ impl DatadogMetricsEncoder {
     }
 }
 
+fn generate_proto_metadata(
+    maybe_pass_through: Option<&DatadogMetricOriginMetadata>,
+    maybe_source_type: Option<&'static str>,
+    origin_product_value: u32,
+) -> Option<ddmetric_proto::Metadata> {
+    generate_origin_metadata(maybe_pass_through, maybe_source_type, origin_product_value).map(
+        |origin| ddmetric_proto::Metadata {
+            origin: Some(ddmetric_proto::Origin {
+                origin_product: origin.product().expect("OriginProduct should be set"),
+                origin_category: origin.category().expect("OriginCategory should be set"),
+                origin_service: origin.service().expect("OriginService should be set"),
```

Comment:
At a high level I agree that its safer to default and log.

But here specifically, if the `generate_origin_metadata()` gave us a Some(origin) , then its a bug if those aren't set. Which actually makes me realize something- in the beginning of the Origin PR I created those functions to set each individual value. But by the end of the PR, we changed the design such that if we set any of the Origin values, we set  all of them. Therefore I feel the functions to set each value should be replaced with a single function to set all. That leaves us less room to accidentally not set one.

I suppose the concern still remains though, so I guess it would still be reasonable to log a warning.

However, all of that said, since this PR hasn't modified any of that, I'd treat this as a tertiary concern and would opt to change that in a separate PR.

WDYT?

Reply:
I'm happy for this to be done in a separate PR 👍🏻

Reply:
👍 
created https://github.com/vectordotdev/vector/pull/18788